Diode Insertion fails if routing obstruction enabled

<!--
    This issue template is REQUIRED for all bug reports. It helps us more
    quickly track, narrow down, and address bugs. 

    Bug reports not adhering to this template will be likely marked invalid and
    closed as there is very little we can do without the requisite information.

    Thank you for understanding!
-->

## Description
If configuration file has `GRT_OBS` and `diode_insertion` stage fails during `run_routing` stage

## Expected behavior
Complete the flow successfully

## Environment
<!--
    This part is incredibly important:

    Please run the following shell command in the OpenLane root folder:

        python3 ./env.py issue-survey

    Then copy and paste the ENTIRE output between the triple-backticks below. (```)

    If the command does not succeed, you are using an out-of-date version of
    OpenLane, and it is recommended that you update.
-->

```
Kernel: Linux v3.10.0-1160.49.1.el7.x86_64
Distribution: centos 7
Python: v3.6.8 (OK)
Container Engine: docker v20.10.16 (OK)
OpenLane Git Version: c40bcaca582bb192606265d49a582fb9ee3ba1be
pip: INSTALLED
python-venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

c40bcac 2022-09-22T07:26:28+02:00 [BOT] Update PDK (#1389) - Openlane Bot -  (HEAD -> master, tag: 2022.09.23, origin/master, origin/HEAD)
c8f917e 2022-09-22T07:23:38+02:00 Correctly decorate wrapper in click_odb (#1381) - Matt Liberty -  ()
f81bd81 2022-09-22T07:21:39+02:00 Correct the installation link (#1384) - Arman Avetisyan -  ()
---
Git Remotes

origin	https://github.com/The-OpenROAD-Project/OpenLane.git (fetch)
origin	https://github.com/The-OpenROAD-Project/OpenLane.git (push)
```

## Reproduction Material
<!--
    You have two options here:

    A. If you see a message like `Reproducible packaged: Please tarball and
       upload <PATH> if you're going to submit an issue` in your logs,
       please also tarball and include that path. The reproducible tarball is
       absolutely required for OpenROAD bugs.

    B. If you don't...
        * Upload a tarball containing the relevant design.
        * List the commands used to run the design.
-->

1. Clone MPW CI: https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI
2. `cd OpenLane-MPW-CI`
3. Update `openlane-tag ` [2022.09.23](https://github.com/The-OpenROAD-Project/OpenLane/releases/tag/2022.09.23)
4. To install latest OpenLane: `./scripts/setup-ci.sh`
5. To reproduce the issue run the design: `./scripts/run-design.sh yonga-100m-ethernet`
Check MPW CI for error: https://github.com/The-OpenROAD-Project/OpenLane-MPW-CI/pull/91/checks?check_run_id=8532111552
Flow will fail during diode insertion stage with following error

## Logs
<!--
    Feel free to add any relevant logs to this section.
    
    Please do ensure they're inside the triple-backticks. (```)
-->
```
[STEP 13]
[INFO]: Running Diode Insertion (log: logs/routing/13-obs.log)...
invalid command name "set_db"
    while executing
"set_db $save_db"
    (procedure "apply_route_obs" line 15)
    invoked from within
"apply_route_obs"
    (procedure "add_route_obs" line 3)
    invoked from within
"add_route_obs"
    (procedure "run_routing" line 20)
    invoked from within
"run_routing"
    (procedure "run_routing_step" line 8)
    invoked from within
"[lindex $step_exe 0] [lindex $step_exe 1] "
    (procedure "run_non_interactive_mode" line 52)
    invoked from within
"run_non_interactive_mode {*}$argv"
    invoked from within
"if { [info exists flags_map(-interactive)] || [info exists flags_map(-it)] } {
    if { [info exists arg_values(-file)] } {
        run_file [file nor..."
    (file "./flow.tcl" line 401)
make: *** [Makefile:175: quick_run] Error 1
make: Leaving directory '/tmp/workspace/_Public_production-small-3_PR-91@2/OpenLane'
```

--- a/scripts/tcl_commands/routing.tcl
+++ b/scripts/tcl_commands/routing.tcl
@@ -297,7 +297,7 @@ proc apply_route_obs {args} {
     puts_verbose "Obstructions added over $::env(GRT_OBS)."
 
     set_def $save_def
-    set_db $save_db
+    set_odb $save_db
 
     TIMER::timer_stop
     exec echo "[TIMER::get_runtime]" | python3 $::env(SCRIPTS_DIR)/write_runtime.py "adding routing obstructions - openlane"

